### DISCUSSION TITLE
NextAuth Fails when deployed on IIS with ReverseProxy

### DISCUSSION THREAD

**Discussion by amaster507** [nextauthjs/next-auth#3328] [Help] [answered]
I have a Next.JS app that I built on the intranet of my employers organization. Everything is Windows based so I had to deploy with IIS locally instead of using a service like Vercel as I have done in the past. I was able to develop the next-auth portion of the app and integrated with Active Directory for authentication and then can use Group Membership to control Authorization to portions of the application as needed. Everything worked perfect during development, then I deployed it to IIS with Reverse Proxy.

I did not setup a custom server.js script for my Next.JS application, I simply run it on the Windows Server using `npm run build && npm start` which has the app running on port 3000. I then setup a site in IIS and configured URL Rewriting as a Reverse Proxy to `http://localhost:3000` and all was good, or so I thought. I was able to authenticate and use the app from multiple machines, but then, I launched it to my department for more test users for a beta launch.

What happened was that a person would authenticate in the application and then that authenticated session/jwt would be given to the next person in line who requested the authentication page and they would be redirected to the homepage of the app with the identity of the other authenticated person.

In essence, I setup session hijacking by using a reverse proxy.

I am stuck to only being able to deploy this Next.JS application on a Windows Server. I do have full admin rights of this server though, so if there is a different way this needs to be configured maybe even without IIS, then I can follow through with that.

I have not been able to find anybody else using Next.JS and next-auth deployed on IIS except for a few tutorials, that are specifically for Azure.

Can anyone help me with this? What other information is needed?

I am using next-auth 3.29.0 and wondering if maybe the upgrade to 4.0 would fix anything, but I don't see anything related to this by skimming the release notes.

**Comment by amaster507** [comment]
Maybe related to #3183 but no responses there yet either.

**Comment by balazsorban44** [comment]
Would this be useful? https://next-auth.js.org/configuration/options#nextauth_url_internal

This helped others in situations with Proxy before.

**Comment by amaster507** [comment, marked as answer]
The url internal variable did not fix the problem, but it also did not break anything worse, so leaving it in place. I believe this will help prevent extra round trips through the front door of IIS where it should just be able to go local host. 

However, I did figure out the problem and found a fix.

The fix was to disable IIS caching. By default caching is turned on with no config options that would do things like only cache if matching header field, etc. For now I just disabled all caching for this site. I will potentially add a better caching config settings to only cache certain routes or cache using header keys for cache splitting.

Thank you for the suggestion @balazsorban44

**Reply by Andrew-Gee** [reply]
How did you 'disable IIS caching'? I've tried disabling output cache, and that had no effect.

**Reply by avenmia** [reply]
I had the same issue and disabled it this way:
In IIS:
* Click on the server
* Application Request Routing
* Server Proxy Settings
* Uncheck Enable disk cache
